libs/src/services/ethContracts/governanceClient.js

```diff
@@ -105,6 +136,160 @@ class GovernanceClient extends ContractClient {
+    functionSignature,
+    callData,
+    description
+  }) {
+    const method = await this.getMethod(
+      'submitProposal',
+      targetContractRegistryKey,
+      callValue,
+      functionSignature,
+      callData,
+      description
+    )
+    const tx = await this.web3Manager.sendTransaction(method, DEFAULT_GAS_AMOUNT * 10)
+    if (tx && tx.events && tx.events.ProposalSubmitted && tx.events.ProposalSubmitted.returnValues) {
+      const id = tx.events.ProposalSubmitted.returnValues.proposalId
+      return id
+    }
+    throw new Error('No proposal Id')
+  }
+
+  async submitVote ({
+    proposalId,
+    vote
+  }) {
+    const method = await this.getMethod(
+      'submitVote',
+      proposalId,
+      vote
+    )
+    await this.web3Manager.sendTransaction(method, DEFAULT_GAS_AMOUNT)
+  }
+
+  async updateVote ({
+    proposalId,
+    vote
+  }) {
+    const method = await this.getMethod(
+      'updateVote',
+      proposalId,
+      vote
+    )
+    await this.web3Manager.sendTransaction(method, DEFAULT_GAS_AMOUNT)
+  }
+
+  async evaluateProposalOutcome (
+    proposalId
+  ) {
+    const method = await this.getMethod(
+      'evaluateProposalOutcome',
+      proposalId
+    )
+    const outcome = await this.web3Manager.sendTransaction(method, DEFAULT_GAS_AMOUNT * 2)
+    return outcome
+  }
+
+  async getVotes ({
+    proposalId,
+    queryStartBlock = 0
+  }) {
+    const contract = await this.getContract()
+    let events = await contract.getPastEvents('ProposalVoteSubmitted', {
+      fromBlock: queryStartBlock,
+      filter: {
+        proposalId: proposalId
+      }
+    })
+    return events.map(formatVote)
+  }
+
+  async getVoteUpdates ({
+    proposalId,
+    queryStartBlock = 0
+  }) {
+    const contract = await this.getContract()
+    let events = await contract.getPastEvents('ProposalVoteUpdated', {
+      fromBlock: queryStartBlock,
+      filter: {
+        proposalId: proposalId
+      }
+    })
+    return events.map(formatVote)
+  }
+
+  async getVoteSubmissionsByAddress ({
+    addresses,
+    queryStartBlock = 0
+  }) {
+    const contract = await this.getContract()
+    let events = await contract.getPastEvents('ProposalVoteSubmitted', {
+      fromBlock: queryStartBlock,
+      filter: {
+        voter: addresses
+      }
+    })
+    return events.map(formatVote)
+  }
+
+  async getVoteUpdatesByAddress ({
+    addresses,
+    queryStartBlock = 0
+  }) {
+    const contract = await this.getContract()
+    let events = await contract.getPastEvents('ProposalVoteUpdated', {
+      fromBlock: queryStartBlock,
+      filter: {
+        voter: addresses
+      }
+    })
+    return events.map(formatVote)
+  }
+
+  // Helpers
+
+  /**
+   * ABI encodes argument types and values together into one encoded string
+   * @param {Web3} web3
+   * @param {Array<string>} types
+   * @param {Array<string>} values
+   */
+  abiEncode (web3, types, values) {
+    return web3.eth.abi.encodeParameters(types, values)
```

Comment:
i think you could just use this.web3Manager.getWeb3() if we're moving this inside the class

Reply:
yeah good call